.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,31 @@
+
+//! need to cleanup, also codespace ran out of space (8 -core 32GB RAM • 64GB works so far without losing space)
+//! has over an hour + upstart and needs further investigation overall
+
+{
+	"name": "UIPA DevBox",
+	"build": {
+	  "context": "..",
+	  "dockerfile": "../Dockerfile"
+	},
+	"runArgs": [
+	  "--init",
```

Comment:
Is there a specific reason we chose to user a Docker in Docker approach here?

Reply:
I think it's to allow starting up the database and search engine to be started from a terminal in the container. The dev container's postStartCommand property causes the `cleanup_and_start.sh` script to run `docker-compose` to start the dev web server, database, and search engine in containers.

The way I read the Dev Containers docs, I'm guessing that using a docker-compose.yml file in the `build` property of the `devcontainer.json` file, the database and search engine could be run automatically along side the dev web server without having to run them all from inside a dev container.